docs/docsite/rst/dev_guide/developing_modules_documenting.rst

```diff
@@ -29,30 +30,93 @@ Include it in your module file like this:
     # ... snip ...
     '''
 
-The ``description``, and ``notes`` fields
-support formatting with some special macros.
 
-These formatting functions are ``U()``, ``M()``, ``I()``, and ``C()``
-for URL, module, italic, and constant-width respectively. It is suggested
-to use ``C()`` for file and option names, and ``I()`` when referencing
-parameters; module names should be specified as ``M(module)``.
+The following fields can be used and are all required unless specified otherwise
+
+* ``module:``
+  The name of the module, same as the file without `.py`.
+* ``short_description:``
+
+  * Displayed on the :doc:`../list_of_all_modules` page and ``ansible-doc -l``.
+  * Must have enough detail without the context of the category.
+  * Without a trailing full stop
+* ``description:``
+  * A longer description than `short_description`, which goes into more detail, generally two or more sentences.
+  * Must be written in full sentences, i.e. with capital letters and fullstops.
+  * Shouldn't mention the name module.
+* ``version_added:``
+  The version of Ansible when the module was added.
+  This is a `string`, and not a float, i.e. ``version_added: "2.1"``
+* ``author:``
+  In the form ``First Last (@GitHubID)``, use a multi-line list if there is more than one.
+* ``options:``
+  One per option name
+
+  * ``description:``
+
+    * Detailed explanation of what this option does and what it does. It should be written in full sentences.
+    * Should not list the option values (that's what ``choices:`` is for, though it should explain `what` the values do if they aren't obvious.
+    * If an argument takes both True/False and Yes/No, the documentation should use True and False.
+    * If an optional parameter is sometimes required this need to be reflected in the documentation, e.g. "Required when I(state=present)."
+    * Mutually exclusive options must be documented as the final sentence on each of the options.
+  * ``required:``
+    Only needed if true, otherwise it is assumed to be false.
+  * ``default:``
+    * If `required` is false/missing, `default` may be specified (assumed 'null' if missing).
+    * Ensure that the default parameter in docs matches default parameter in code. The default option must not be listed as part of the description.
```

Comment:
Ensure that the default parameter in the docs matches default parameter in the code.